piqe_ocp_lib/api/resources/ocp_machine_management.py

```diff
@@ -353,3 +354,128 @@ def is_machine_deleted(self, machine_name: str, timeout: int = TIMEOUT) -> bool:
+        self.machinehealthcheck = self.dyn_client.resources.get(api_version=self.api_version, kind=self.kind)
+        self.machineset = OcpMachineSet(kube_config_file=kube_config_file)
+        self.machine = OcpMachines(kube_config_file=kube_config_file)
+        self.node = OcpNodes(kube_config_file=kube_config_file)
+
+    def get_all_machine_health_checks(self) -> ResourceList:
+        """
+        Get all defined Machine health check
+        :return: MachineHealthCheck on success OR an empty list on failure
+        """
+        api_response = list()
+        try:
+            api_response = self.machinehealthcheck.get(namespace=MACHINE_NAMESPACE)
+        except ApiException as e:
+            logger.error("Exception while getting all Machine Health Checks: %s\n", e)
+            raise Exception(e)
+        return api_response
+
+    def get_machine_health_check(self, machine_health_check_name: str) -> ResourceInstance:
+        """
+        Get a Machine health check set by name
+        :param machine_health_check_name: (str) name of the machine set
+        :return: MachineHealthCheck object on success OR None on failure
+        """
+        api_response = None
+        try:
+            api_response = self.machinehealthcheck.get(name=machine_health_check_name, namespace=MACHINE_NAMESPACE)
+        except ApiException as e:
+            logger.error("Exception while getting Machine Health Check: %s\n", e)
+            raise Exception(e)
+        return api_response
+
+    def configure_machine_health_check(self, machine_health_check_name: str) -> bool:
+        """
+        Create Machine Health Check for a machineset.
+        :param machine_health_check_name: (str) name of the machine health check
+        :return: (bool) True when successfully configured a Machine Heath Check object OR False otherwise
+        """
+        mhc_yaml = """
+        apiVersion: machine.openshift.io/v1beta1
+        kind: MachineHealthCheck
+        metadata:
+          name: <mhc>
+          namespace: openshift-machine-api
+        spec:
+          selector:
+            matchLabels:
+              machine.openshift.io/cluster-api-machineset: <cluster_name>-<label>-<zone> 
+          unhealthyConditions:
+          - type:    "Ready"
+            timeout: "300s" 
+            status: "False"
+          - type:    "Ready"
+            timeout: "300s" 
+            status: "Unknown"
+          maxUnhealthy: "40%" 
+          nodeStartupTimeout: "10m" 
+        """
+        mhc_data = yaml.load(mhc_yaml, Loader=yaml.SafeLoader)
+        machine_set_list = self.machineset.get_machine_sets()
+        machine_set_name = machine_set_list.items[0].metadata.name
+        mhc_data["metadata"]["name"] = f"{machine_health_check_name}"
+        mhc_data["spec"]["selector"]["matchLabels"]["machine.openshift.io/cluster-api-machineset"] = machine_set_name
+
+        def _create_machine_health_check(self, machine_set_name: str, mhc_data) -> ResourceInstance:
```

Comment:
I'd like to chat and discuss the choice of making the create/delete/watch methods private/embedded in the configure method.